colossalai/inference/engine/modeling/chatglm2.py

```diff
@@ -279,15 +292,21 @@ def chatglm_encoder_forward(
             )
             infer_state.decode_layer_id += 1
 
-            hidden_states, _ = layer_ret
+            hidden_states, next_kv_cache = layer_ret
+            if use_cache:
+                next_decoder_cache += (next_kv_cache,)
 
         hidden_states = hidden_states.transpose(0, 1).contiguous()
 
-        if self.post_layer_norm and (stage_manager.is_last_stage() or stage_manager.num_stages == 1):
+        if self.post_layer_norm and (
+            stage_manager is None or stage_manager.is_last_stage() or stage_manager.num_stages == 1
```

Comment:
if num_stages==1 then stage manager should be `None`?

Reply:
done